lib/internal/url.js

```diff
@@ -400,7 +400,9 @@ Object.defineProperties(URL.prototype, {
           ret += '@';
         }
         ret += options.unicode ?
-          domainToUnicode(this.host) : this.host;
+          domainToUnicode(this.hostname) : this.hostname;
+        if (ctx.port !== '')
```

Comment:
I think this should be `ctx.port !== null`?

Reply:
The WAHTWG URL API uses `''` as the empty value, not `null`.

Reply:
```javascript
> new url.URL("https://nodejs.org/en/")
URL {
  href: 'https://nodejs.org/en/',
  origin: 'https://nodejs.org',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'nodejs.org',
  hostname: 'nodejs.org',
  port: '',
  pathname: '/en/',
  search: '',
  searchParams: URLSearchParams {},
  hash: '' }
```

Reply:
Oh my bad! This is the internal context...